components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImpl.java

```diff
@@ -120,12 +120,14 @@ public void insertAuthorizationCode(String authzCode, String consumerKey, String
             //insert the hash value of the authorization code
             prepStmt.setString(13, getHashingPersistenceProcessor().getProcessedAuthzCode(authzCode));
             prepStmt.setString(14, getPersistenceProcessor().getProcessedClientId(consumerKey));
+            int appTenantId = IdentityTenantUtil.getTenantId(appTenantDomain);
             if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
                 prepStmt.setString(15, authenticatedIDP);
-                prepStmt.setInt(16, tenantId);
-                prepStmt.setInt(17, IdentityTenantUtil.getTenantId(appTenantDomain));
+                // Set tenant ID of the IDP by considering it is same as appTenantID.
+                prepStmt.setInt(16, appTenantId);
+                prepStmt.setInt(17, appTenantId);
```

Comment:
The related SQL query.

<img width="803" alt="Screenshot 2023-10-15 at 15 16 51" src="https://github.com/wso2-extensions/identity-inbound-auth-oauth/assets/35717390/c7f5580d-9e26-4b3f-b8da-421897587f06">